File: merte/python/10.carpma.kullanmadan.carpma.py
x= int(input('1. Sayi: '))
y= int(input('2. Sayi: '))

flag = '+'
if x<0 and y<0: 
  flag = '-'
  x= -x
if y<0: y, x= x, y

carpim=0
for i in range(y,0,-1):
    carpim+=x

if flag =='-':
    carpim = -carpim
print(carpim)

File: merte/cpp/10.carpma.kullanmadan.carpma.cpp
#include <iostream>

using namespace std;

int main() {
    float n1;
    float n2;
    float toplam=0;
    cout << "Birinci sayi: ", cin >> n1;
    cout << "Ikinci sayi: ", cin >> n2;
    bool flag = (n1>n2) ? true : false;
    //TODO:
    /* 0-1 arası negatif ve pozitif
    sayılarda çarpım sorunu var */

    if (flag && n1>0){
        for (int i=0; i<n1; i++){
            toplam+=n2;
        }
    } else if (!flag && n2>0){
        for (int i=0; i<n2; i++){
            toplam+=n1;
        }
    } else if (flag && n1<0){
        for (int i=0; i>n1; i--){
            toplam+=n2;
        }
    } else if (!flag && n2<0){
        for (int i=0; i>n2; i--){
            toplam+=n1;
        }
    }
    if (n1 < 0 && n2 < 0) toplam *= -1;
    cout << toplam;
}